### PR TITLE
Update CircleCI to start building with Elixir 1.7

### PR DESCRIPTION
**Pull request by fhunleth**
Merge the ex_docs PR first

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/elixir:1.6.4
+      - image: circleci/elixir:1.7
```
Comment (GregMefford) [inline .circleci/config.yml:8]: Why not `1.7.3`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -15,4 +15,6 @@ jobs:
       - run: mix local.hex --force && mix local.rebar --force
       - run: mix deps.get --only test
       - run: mix compile
+      - run: mix hex.build
       - run: mix format --check-formatted
+      - run: mix docs
```
Comment (GregMefford) [inline .circleci/config.yml:20]: Is this just to make sure it doesn't crash?

**Review by GregMefford — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (fhunleth) [reply .circleci/config.yml:8]: It doesn't exist yet. :(

File: .circleci/config.yml (hunk 2 above)
Comment (fhunleth) [reply .circleci/config.yml:20]: yep

**Pull request merged**